skbio/sequence/_sequence.py

```diff
@@ -526,7 +572,9 @@ def concat(cls, sequences, how='strict'):
         pm = pd.concat(pm_data, join=how, ignore_index=True)
         bytes_ = np.concatenate(seq_data)
 
-        return cls(bytes_, positional_metadata=pm)
+        im = IntervalMetadata.concat(i.interval_metadata for i in seqs)
```

Comment:
there is nothing special about the sequence without `interval_metadata`, because the getter will return the empty `IntervalMetadata` with proper upper_bound. The existing code works alright.